src/lib/y2users/linux/writer.rb

```diff
@@ -158,6 +165,68 @@ def add_user(user, issues)
         log.error("Error creating user '#{user.name}' - #{e.message}")
       end
 
+      # Command for creating new users
+      GROUPADD = "/usr/sbin/useradd".freeze
+      private_constant :USERADD
+
+      # Executes the command for creating the group
+      #
+      # @param group [Y2User::Group] the group to be created on the system
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def add_group(group, issues)
+        args = [GROUPADD]
+        args << "--gid" << group.gid if group.gid
+        # TODO: system groups?
+        Yast::Execute.on_target!(args)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          format(_("The user '%{username}' could not be created"), username: user.name)
+        )
+        log.error("Error creating user '#{user.name}' - #{e.message}")
+      end
+
+      USERMOD_ATTRS = [:home, :shell, :gecos].freeze
+
+      def modify(new_user, old_user, issues)
+        change_password(new_user, issues) if new_user.password != old_user.password
+
+        usermod_changes = USERMOD_ATTRS.any? do |attr|
+          !new_user.public_send(attr).nil? &&
+            (new_user.public_send(attr) != old_user.public_send(attr))
+        end
+        # usermod also manage suplimentary groups, so compare also them
+        usermod_changes ||= different_groups?(new_user, old_user)
+        usermod_modify(new_user, old_user, issues) if usermod_changes
+      end
+
+      def different_groups?(lhu, rhu)
+        lhu.groups(with_primary: false).sort != rhu.groups(with_primary: false).sort
+      end
+
+      USERMOD = "/usr/sbin/usermod".freeze
+      private_constant :USERMOD
+      def usermod_modify(new_user, old_user, issues)
+        args = [USERMOD]
+        if new_user.gecos != old_user.gecos && new_user.gecos
```

Comment:
I think the last part of the condition is not needed since `gecos` should be always an array. Emptied or filled array, but not nil.

Reply:
yeah, probably something to change as it is different if you intentionally keep it empty or use nil for I do not modify.

Reply:
@ancorgs ^^^